library/analytics/src/main/java/org/cru/godtools/analytics/firebase/FirebaseAnalyticsService.kt

```diff
@@ -54,54 +65,51 @@ class FirebaseAnalyticsService @VisibleForTesting internal constructor(
     @MainThread
     @Subscribe(threadMode = ThreadMode.MAIN_ORDERED)
     fun onAnalyticsEvent(event: AnalyticsBaseEvent) {
-        if (event.isForSystem(AnalyticsSystem.FIREBASE)) when (event) {
+        if (event.isForSystem(AnalyticsSystem.FIREBASE) || event.isForSystem(AnalyticsSystem.ADOBE)) when (event) {
             is AnalyticsScreenEvent -> handleScreenEvent(event)
             is AnalyticsActionEvent -> handleActionEvent(event)
         }
     }
 
-    // region ActivityLifecycleCallbacks
-    init {
-        app.registerActivityLifecycleCallbacks(this)
-    }
-
-    private var currentActivity: Activity? by weak()
-
-    override fun onActivityResumed(activity: Activity) {
-        currentActivity = activity
-    }
-
-    override fun onActivityPaused(activity: Activity) {
-        currentActivity = currentActivity?.takeUnless { it === activity }
-    }
-
-    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = Unit
-    override fun onActivityStarted(activity: Activity) = Unit
-    override fun onActivityStopped(activity: Activity) = Unit
-    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) = Unit
-    override fun onActivityDestroyed(activity: Activity) = Unit
-    // endregion ActivityLifecycleCallbacks
-    // endregion Tracking Events
-
     @MainThread
     private fun handleScreenEvent(event: AnalyticsScreenEvent) {
-        currentActivity?.let { firebase.setCurrentScreen(it, event.screen, null) }
+        val bundle = Bundle().apply {
+            putString(FirebaseAnalytics.Param.SCREEN_NAME, event.screen)
+            putString(USER_PROP_APP_NAME, VALUE_GODTOOLS)
+            event.locale?.let { putString(USER_PROP_CONTENT_LANGUAGE, LocaleCompat.toLanguageTag(it)) }
+            event.adobeSiteSection?.let { putString(SITE_SECTION, event.adobeSiteSection) }
+            event.adobeSiteSubSection?.let { putString(SITE_SUB_SECTION, event.adobeSiteSubSection) }
+        }
+        firebase.logEvent(FirebaseAnalytics.Event.SCREEN_VIEW, bundle)
     }
 
     @MainThread
     private fun handleActionEvent(event: AnalyticsActionEvent) {
-        firebase.logEvent(event.firebaseEventName, null)
+        val bundle = Bundle().apply {
+            event.adobeAttributes?.forEach { attribute ->
+                val attributeKey = attribute.key.replace(Regex("[ \\-.]"), "_")
+                putString(attributeKey, attribute.value.toString())
+            }
+        }
+
+        firebase.logEvent(event.firebaseEventName, bundle)
     }
 
     init {
-        oktaUserProfileProvider.userInfoFlow()
-            .map { it?.ssoGuid }
-            .distinctUntilChanged()
-            .onEach { firebase.setUserId(it) }
+        oktaUserProfileProvider.userInfoFlow(refreshIfStale = false)
+            .onEach { it ->
+                firebase.setUserProperty(USER_PROP_LOGGED_IN_STATUS, "${it != null}")
+                firebase.setUserProperty(USER_PROP_APP_NAME, VALUE_GODTOOLS)
```

Comment:
with setting the app name here, do you also need to set it on [line 78](https://github.com/CruGlobal/godtools-android/pull/1396/files#diff-16cc0dd8554c9cc5ad5d84d969fb2c391765505cce3113ebc3537609ffe13763R78)?

Reply:
That's what they requested, but I'm confirming that they realize we are sending it twice.

Reply:
They said just to leave it for now.